mito-ai/mito_ai/tests/providers_test.py

```diff
@@ -0,0 +1,167 @@
+import os
+from datetime import datetime
+from unittest.mock import patch
+from mito_ai.providers import OpenAIProvider
+
+CHAT_LIMIT = 500
+
+
+def test_os_user_mito_server_below_limit():
+    """
+    Open source user, with no OpenAI API key set (Mito server), below both limits.
+    No error should be thrown.
+    """
+
+    llm = OpenAIProvider()
+    today = datetime.now().strftime("%Y-%m-%d")
+
+    with (
+        patch.dict(os.environ, {"OPENAI_API_KEY": ""}),
+        patch("mito_ai.providers._num_usages", 1),
+        patch("mito_ai.providers._first_usage_date", today),
+    ):
+        capabilities = llm.capabilities
+
+        assert capabilities.provider == "Mito server"
+        assert llm.last_error is None
```

Comment:
Hmmm maybe we should also be testing the `check_mito_server_quota` function directly? Cause that is the actual function that is stopping us from using the server, right? Not the capabilities display.